### DISCUSSION TITLE
GitHub Action workflow is executed for a PR from a forked repo if the PR changes the workflow file

### DISCUSSION THREAD

**Discussion by huynguyendev** [community/community#25198] [Actions]
<p>According to <a href="https://github.community/t/run-a-github-action-on-pull-request-for-pr-opened-from-a-forked-repo/16054/2">this thread</a>, actions of a PR from a fork repo will be executed against the fork, and only if the fork is in the beta program.</p>
<p>I tested this by creating a test account (<a class="mention" href="/huynguyendev">@huynguyendev</a>) that is not in the beta. I then opened up a <a href="https://github.com/TextureGroup/Texture/pull/1628/checks" rel="nofollow noopener">pull request</a> to TextureGroup/Texture (which is in the beta) from my forked repo. The commit changes Texture’s worflow file to add pull_request event and voila, the workflow is now executed on my PR.</p>
<p>This has the potential of allowing bad actors to run their own scripts by simply opening up PRs on open source projects. Can someone verify if this is in fact intended behavior?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>It allows bad actors to run their own scripts, but only in the context of their own repository. Basically, it works like this:</p>
<ol>
<li>A repo named <code>foo/bar</code> exists</li>
<li>I fork the repository <code>foo/bar</code> to <code>lee-dohm/bar</code>
</li>
<li>I make a change to a workflow on a branch in <code>lee-dohm/bar</code> and open a pull request to <code>foo/bar</code>
</li>
</ol>
<p>Leaving aside the “which repo is in the Actions beta” part, the workflow executes in <code>lee-dohm/bar</code> even though the result is visible in <code>foo/bar</code>. This means that I <strong>can’t</strong> create a workflow that emails the <code>secrets</code> of <code>foo/bar</code> to me by creating a PR because it would email me the <code>secrets</code> of <code>lee-dohm/bar</code>.</p>
<p>I hope that helps to clarify things.</p>

**Reply by amithm7** [reply]
> the workflow executes in lee-dohm/bar even though the result is visible in `foo/bar`.

That wouldn't be the correct phrase to describe this behaviour.

The workflow from fork does execute in original repo. `foo/bar` as "Fork PR workflow" but with read-only permissions. No access to original repo. secrets, etc.

See: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

**Comment by s-i-l-k-e** [comment]
<p>Is this still the case?</p>

**Comment by ethomson** [comment]
<p>This is still the case.  Forked repositories only have a read-only version of the repository token so that they can do a clone, and do not get a copy of workflow secrets.</p>

**Comment by domoritz** [comment]
<p>Builds should still be triggered even in forks, right? Somehow I don’t see any builds for a fork in this PR </p>
<p><a href="https://github.com/vega/vega-lite/pull/5438" rel="nofollow noopener">https://github.com/vega/vega-lite/pull/5438</a>. Is there a setting to enable builds on forks? </p>

**Comment by derberg** [comment]
<p>Hi <a class="mention" href="/lee-dohm">@lee-dohm</a>  and <a class="mention" href="/ethomson">@ethomson</a> </p>
<p>This limited support for forks is a big adoption problem, I hope you see it.</p>
<p>I wrote a blog post about it where I not only complain but also suggest a very handy possible solution that you could copy from other tools - <a href="https://dev.to/derberg/github-actions-when-fascination-turns-into-disappointment-4d75" rel="nofollow noopener">https://dev.to/derberg/github-actions-when-fascination-turns-into-disappointment-4d75</a>. I hope this helps, somehow.</p>

**Comment by ethomson** [comment]
<p>As we mentioned in other threads that you’re in - this is something that we’re investigating and working on.</p>
<p>In the meantime, you’re welcome to implement exactly the proposal that you describe in your blog post.  You can trigger a workflow on an issue comment, examine that issue comment for <code>ok-to-test</code>, then do whatever is necessary with secrets.  They’ll be available to that workflow since it’s not running on a pull request event from a fork.</p>

**Comment by derberg** [comment]
<p>Interesting. But how would I integrate it with pull requests?</p>
<p>For example. I need to enable static code analysis with eslint, and want it on PR of course.</p>
<p>So I would have eslint as part of the workflow that is triggering on issue comment? and I would somehow reference the PR number in the issue comment, so the workflow knows what repo and branch to checkout?</p>

**Comment by ibakshay** [comment]
<p><a class="mention" href="/ethomson">@ethomson</a>, We are also facing the adoption problems for quite some time (nearly 5 months). Our GitHub Action progress is in a standstill due to this limitation.  </p>
<p>Can you give us an ETA ? So that I can pass it on to my colleagues.</p>
